src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -92,13 +92,15 @@ internal sealed class HttpEnvironmentProxy : IWebProxy
         private string[] _bypass = null;// list of domains not to proxy
         private ICredentials _credentials;
 
-        public static HttpEnvironmentProxy TryToCreate()
+        public static bool TryCreate(out IWebProxy proxy)
         {
             // Get environmental variables. Protocol specific take precedence over
             // general all_*, lower case variable has precedence over upper case.
             // Note that curl uses HTTPS_PROXY but not HTTP_PROXY.
             // For http, only http_proxy and generic variables are used.
 
+            proxy = null;
```

Comment:
It'd be better to move this down to just before you return false, so that essentially the logic is:
```C#
if (httpProxy != null || httpsProxy != null)
{
    proxy = new HttpEnvironmentProxy(...);
    return true;
}
else
{
    proxy = null;
    return false;
}
```